libc/src/math/generic/expm1.cpp

```diff
@@ -0,0 +1,493 @@
+//===-- Double-precision e^x - 1 function ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/math/expm1.h"
+#include "common_constants.h" // Lookup tables EXP_M1 and EXP_M2.
+#include "explogxf.h"         // ziv_test_denorm.
+#include "src/__support/CPP/bit.h"
+#include "src/__support/CPP/optional.h"
+#include "src/__support/FPUtil/FEnvImpl.h"
+#include "src/__support/FPUtil/FPBits.h"
+#include "src/__support/FPUtil/PolyEval.h"
+#include "src/__support/FPUtil/double_double.h"
+#include "src/__support/FPUtil/dyadic_float.h"
+#include "src/__support/FPUtil/except_value_utils.h"
+#include "src/__support/FPUtil/multiply_add.h"
+#include "src/__support/FPUtil/nearest_integer.h"
+#include "src/__support/FPUtil/rounding_mode.h"
+#include "src/__support/FPUtil/triple_double.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/optimization.h" // LIBC_UNLIKELY
+
+#include <errno.h>
+
+// #define DEBUGDEBUG
```

Comment:
Did you mean to leave the debug code in here?

Reply:
I found myself retyping these debug codes quite many times, and I couldn't find a better place to share this type of debugging codes yet.  So I think I should leave it at least in one function so that a simple copy paste during development of other functions would be enough.

Reply:
SGTM